src/plugins/robot-memory/event_trigger_manager.h

```diff
@@ -62,39 +55,43 @@ class EventTriggerManager
      */
 	template <typename T>
 	EventTrigger *
-	register_trigger(mongo::Query query,
-	                 std::string  collection,
-	                 void (T::*callback)(mongo::BSONObj),
+	register_trigger(bsoncxx::document::value query,
+	                 std::string              collection,
+	                 void (T::*callback)(bsoncxx::document::value),
 	                 T *obj)
 	{
 		//lock to be thread safe (e.g. registration during checking)
 		fawkes::MutexLocker lock(mutex_);
 
 		//construct query for oplog
-		mongo::BSONObjBuilder query_builder;
-		query_builder.append("ns", collection);
+		namespace basic_builder = bsoncxx::builder::basic;
+		auto oplog_query        = basic_builder::document{};
+		oplog_query.append(basic_builder::kvp("ns", collection));
 		// added/updated object is a subdocument in the oplog document
-		for (mongo::BSONObjIterator it = query.getFilter().begin(); it.more();) {
-			mongo::BSONElement elem = it.next();
-			query_builder.appendAs(elem, std::string("o.") + elem.fieldName());
+		auto filter = query.view()["$filter"].get_document().view();
+		for (auto &&it = filter.begin(); it != filter.end(); it++) {
+			auto elem = *it;
+			oplog_query.append(
+			  basic_builder::kvp(std::string("o.") + std::string(elem.key()), elem.get_value()));
 		}
-		mongo::Query oplog_query = query_builder.obj();
-		oplog_query.readPref(mongo::ReadPreference_Nearest, mongo::BSONArray());
+		//mongo::Query oplog_query = query_builder.obj();
+		// TODO: make sure we set the read preference of the client
```

Comment:
Is this currently not possible or have there been problems, i.e., why is this still a TODO?

Reply:
Fixed (on client level) in dcb75fcc4048c97b87353bef2c85adaeb209ecbf